src/field_theory/cardinality.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import ring_theory.localization.cardinality
+import algebra.is_prime_pow
+import field_theory.finite.galois_field
+import data.equiv.transfer_instance
+import algebra.ring.ulift
+import data.mv_polynomial.cardinal
+import data.rat.denumerable
+
+/-!
+# Cardinality of Fields
+
+In this file we show all the possible cardinalities of fields. All infinite cardinals can harbour
+a field structure, and so can all types with prime power cardinalities, and this is sharp.
+
+## Main statements
+
+* `fintype.nonempty_field_iff`: A `fintype` can be given a field structure iff its cardinality is a
+  prime power.
+* `infinite.nonempty_field` : Any infinite type can be endowed a field structure.
+
+-/
+
+local notation `‖` x `‖` := fintype.card x
+
+open_locale cardinal non_zero_divisors
+
+universe u
+
+/-- A `fintype` can be given a field structure iff its cardinality is a prime power. -/
+lemma fintype.nonempty_field_iff {α} [fintype α] :
+  nonempty (field α) ↔ is_prime_pow (‖α‖) :=
+begin
+  split,
+  swap,
+  { rintros ⟨p, n, hp, hn, hα⟩,
+    haveI := fact.mk (nat.prime_iff.mpr hp),
+    exact ⟨(fintype.equiv_of_card_eq ((galois_field.card p n hn.ne').trans hα)).symm.field⟩ },
+  rintro ⟨h⟩,
+  casesI char_p.exists α with p _,
+  haveI hp := fact.mk (char_p.char_is_prime α p),
+  let b := is_noetherian.finset_basis (zmod p) α,
+  rw [module.card_fintype b, zmod.card, is_prime_pow_pow_iff],
+  { exact hp.1.is_prime_pow },
+  rw ←finite_dimensional.finrank_eq_card_basis b,
+  exact finite_dimensional.finrank_pos.ne'
+end
+
+lemma is_prime_pow_card_finite_field {α} [fintype α] [field α] : is_prime_pow (‖α‖) :=
+fintype.nonempty_field_iff.mp ⟨‹field α›⟩
```

Comment:
In my opinion it would be cleaner to prove these in the other order

Reply:
Oh! That's a good idea, yes